bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/JSONResponse.java

```diff
@@ -64,79 +102,70 @@ private static ResponseBuilder response(Response.Status status) {
             // return the existing object
-            ret.add("entity", GSON.toJsonTree(entity));
+            resultJson.add(JSON_KEY_ENTITY, GSON.toJsonTree(entity));
         }
 
         // is there an exception?
-        if (null != ex) {
-
+        if (ex != null) {
             // JSONify the Exception
-            JsonObject exc = new JsonObject();
-            err.add("exception", exc);
-            {
-                exc.addProperty("class", ex.getClass().getName());
-                exc.addProperty("message", ex.getMessage());
-                exc.addProperty("localized-message", ex.getLocalizedMessage());
-                exc.addProperty("cause", null != ex.getCause() ? ex.getCause().getClass().getName() : null);
-
-                if (WITH_STACKTRACE) {
-                    exc.add("stacktrace", GSON.toJsonTree(ex.getStackTrace()));
-                }
-            }
+            JsonObject exceptionJson = new JsonObject();
+            exceptionJson.addProperty("class", ex.getClass().getName());
+            exceptionJson.addProperty("message", ex.getMessage());
+            exceptionJson.addProperty("localized-message", ex.getLocalizedMessage());
+            exceptionJson.addProperty("cause", null != ex.getCause() ? ex.getCause().getClass().getName() : null);
+            errorJson.add("exception", exceptionJson);
         }
 
-        return ret;
+        return resultJson;
     }
 
-    /**
-     * in case of error (404 and such)
-     *
-     * @param status
-     * @param errormessage
-     * @return Response containing a status and the errormessage in JSON format
-     */
-    public static Response createErrorResponse(Response.Status status, String errormessage) {
-        return createResponse(status, null, errormessage);
+    private Response createErrorResponse(Response.Status status, Object entity, String errormessage) {
+        ResponseBuilder rp = responseBuilder(status);
+        JsonElement errorJson = createErrorJson(errormessage, status, entity, null);
+        rp.entity(errorJson);
+        return rp.build();
     }
 
-    /**
-     * Depending in the status, create a Response object containing either the entity alone or an error JSON
-     * which might hold the entity as well.
-     *
-     * @param status
-     * @param entity
-     * @param errormessage an optional error message (may be null), ignored if the status family is successful
-     * @return Response configure for error or success
-     */
-    public static Response createResponse(Response.Status status, Object entity, String errormessage) {
-        JsonElement ret;
-        if (status.getFamily() == Response.Status.Family.SUCCESSFUL) {
-            // create non-null JsonElement if null!=entity
-            ret = null != entity ? GSON.toJsonTree(entity) : null;
-        } else {
-            ret = createErrorJson(errormessage, status, entity, null);
-        }
+    private Response createResponse(Status status, Object entity) {
```

Comment:
I will push an update which closes `out` when the writing thread finishes. The `in` stream will be GC'ed once the receiving side did consume the response. We can not actively close it while the receiver is reading from it. It will notice the close of `out` though and free up.